dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-completed-fee-record-corrections.controller/helpers/map-reasons-and-values-to-formatted-values.ts

```diff
@@ -0,0 +1,53 @@
+import { CorrectionValuesPartialEntity, getFormattedMonetaryValue, RECORD_CORRECTION_REASON, RecordCorrectionReason } from '@ukef/dtfs2-common';
+import { validateRequiredCorrectionField } from '../../../../../../helpers';
+
+/**
+ * Gets a formatted string value for a given correction reason from the correction values.
+ *
+ * @remarks
+ *
+ * No validation is performed for the {@link RECORD_CORRECTION_REASON.OTHER}
+ * reason as this value is not stored on the entity object. However, it still
+ * needs to be included in the formatted string output with a '-' placeholder
+ * so that the {@link RECORD_CORRECTION_REASON.OTHER} reference in the
+ * 'reasons' column has a corresponding value.
+ *
+ * @param correctionValues - The correction values entity.
+ * @param reason - The correction reason to get the formatted value for.
+ * @returns The formatted value for the correction reason.
+ * @throws Error if the correction reason is unknown.
+ * @throws Error if a required correction value is missing for the given reason.
+ */
+export const getFormattedCorrectionValueForCorrectionReason = (correctionValues: CorrectionValuesPartialEntity, reason: RecordCorrectionReason): string => {
```

Comment:
I did what we did for report period which is to also introduce a separate type that matches the partial entity type called `RecordCorrectionValues` which we could use here. I don't really know what is more correct but it is I think nicer to work with for functions like this perhaps?